config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path.yaml

```diff
@@ -15,10 +15,89 @@ presubmits:
         command:
         - runner.sh
         args:
-        - ./.prow.sh
+        - /bin/sh
+        - -c
+        # Not all branches have Prow support.
```

Comment:
You can target jobs to specific branches, like [here](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-storage/sig-storage-gce-config.yaml#L6)

Reply:
I intentionally want this for all branches, to test that it doesn't fail when applied to branches which don't have the .prow.sh file. This is something that the job has to handle, because I don't think we want to update the Prow config each time we create a new branch.

Reply:
I think every new branch we create from here should support prow.  So we only need to exclude the fixed number of branches that we don't support (using skip_branch).

Reply:
Can we add that as an optimization later?

Reply:
I'm kind of confused actually. The hostpath repo only has a master branch, which should support this right?  So there are currently no branches where this won't work.

Reply:
"master" currently doesn't have support for Prow. It gets added by https://github.com/kubernetes-csi/csi-driver-host-path/pull/16, which is work-in-progress.

So right now, that presubmit job only does something when testing that PR.

Reply:
Why don't we leave it off by default until we have stabilized the job, submitted it, and then we can turn them all on?  I'm not sure I see the purpose of having this job running that doesn't do anything.

Reply:
My plan for rolling our Prow support in all repos is this:
- for each repo, enable a presubmit job with the if check to avoid breaking current PRs
- create a PR similar to https://github.com/kubernetes-csi/csi-driver-host-path/pull/16 for each repo
- if that PR successfully gets tested in Prow, merge it
- if not, keep fixing whatever is broken until it works

From then on, all other PRs also get tested by Prow because the necessary configuration will be in the master branch. Using this approach there's no flag day where we need to halt all development while we role out Prow support everywhere.

Reply:
> I'm not sure I see the purpose of having this job running that doesn't do anything.

The purpose is to test that this incremental rollout will work as intended.

The reason for `always_run: true` is that I got tired of manually starting the job for the WIP PR. If you are worried about wasting compute resources by unnecessarily starting this PR, then I can go back to `always_run: false`.

Reply:
I'm not sure we receive that much PR traffic in the repos to see significant benefit.  With this approach of skipping if it doesn't exist, we get false signals that something is working when it actually isn't.

Reply:
We can roll out incrementally and later make it a hard requirement. How about that?

Reply:
I still think it gives a false sense of security.  Someone is going to have a PR showing that the job passed when the job didn't actually run what it was supposed to.

Reply:
It would be nice if there was some way to note a job as being skipped.  Unsure if prow can support that?

Reply:
We could generate a JUnit file with a single skipped test. But the result posted to Github then would still be just a "success", so that probably doesn't address your concern.

> I still think it gives a false sense of security.

We just need to be careful with how we communicate the rollout. "Everything is checked by Prow now" needs to be avoided. Instead we need to say something like "once the Prow PR in your component is merged, Prow is active". We can also turn off reporting of the job result during the rollout, then no-one is going to be misled by seeing Prow results posted to Github. But then we need to be careful to check the result for the Prow PR manually before merging.

People have been making changes without E2E testing for a while now. I would hope that they don't suddenly become reckless and expect automated testing to catch all mistakes.

Reply:
I would prefer that we don't report results if it's not actually running the tests.  Not all contributors are attending our status meetings.  Even as reviewers/approvers, on every PR, we're going to have to check/remember the status of jobs to see if the tests actually ran or not.  I'd rather not have a false signal available to us.

Reply:
Because of `skip_report: true`, no-one is going to see job results until we change that, which we can do once everything really is in place.

Reply:
@msau42 I've turned off  `always_run`. Can this be merged now?

Reply:
I guess I missed my opportunity to get this merged for today, so I made further changes in this PR:
- `.prow.sh` required, old branches blacklisted, `always_run: false`, `skip_report: false`: I think that's what @msau42 wanted. We can add Prow support to master of each component repo in this mode by manually running the test before merging the "add Prow support" PRs. Once that is completed, we can switch the job on (`always_run: true`).
- periodic testing on Kubernetes 1.13.x
- define the job for all repos: I want to try out the job against more components and in at least one (probably cluster-driver-registrar) also run some custom E2E test.